Src/WindowsServer/WindowsServer.Shared.Tests/FirstChanceExceptionStatisticsTelemetryModuleTest.cs

```diff
@@ -241,6 +241,129 @@ public void FirstChanceExceptionStatisticsTelemetryModuleWillNotDimCapTheSameOpe
         }
 
         [TestMethod]
+        public void FirstChanceExceptionStatisticsTelemetryModuleDoNotIncrementOnRethrow()
+        {
+            var metrics = new List<KeyValuePair<Metric, double>>();
+            this.configuration.MetricProcessors.Add(new StubMetricProcessor()
+            {
+                OnTrack = (m, v) =>
+                {
+                    metrics.Add(new KeyValuePair<Metric, double>(m, v));
+                }
+            });
+
+            using (var module = new FirstChanceExceptionStatisticsTelemetryModule())
+            {
+                module.Initialize(this.configuration);
+
+                try
+                {
+                    try
+                    {
+                        throw new Exception("test");
+                    }
+                    catch (Exception ex)
+                    {
+                        Assert.Equal("test", ex.Message);
+                        throw;
+                    }
+                }
+                catch (Exception exc)
+                {
+                    Assert.Equal("test", exc.Message);
+                }
+            }
+
+            Assert.Equal(2, metrics.Count);
+            Assert.Equal("Exceptions Thrown", metrics[0].Key.Name);
+
+            Assert.Equal(1, metrics[0].Value, 15);
+            Assert.Equal(0, metrics[1].Value, 15);
+        }
+
+        [TestMethod]
+        public void FirstChanceExceptionStatisticsTelemetryModuleWasTrackedReturnsTrueForTheSameException()
+        {
+            using (var module = new FirstChanceExceptionStatisticsTelemetryModule())
+            {
+                module.Initialize(this.configuration);
+
+                var exception = new Exception();
+
+                Assert.False(module.WasExceptionTracked(exception));
+                Assert.True(module.WasExceptionTracked(exception));
+                Assert.True(module.WasExceptionTracked(exception));
+                Assert.True(module.WasExceptionTracked(exception));
+            }
+        }
+
+        [TestMethod]
+        public void FirstChanceExceptionStatisticsTelemetryModuleWasTrackedReturnsTrueForInnerException()
+        {
+            using (var module = new FirstChanceExceptionStatisticsTelemetryModule())
+            {
+                module.Initialize(this.configuration);
+
+                var exception = new Exception();
+
+                Assert.False(module.WasExceptionTracked(exception));
+
+                var wrapper = new Exception("wrapper", exception);
+
+                Assert.True(module.WasExceptionTracked(wrapper));
+                Assert.True(module.WasExceptionTracked(wrapper));
```

Comment:
Was the intent to check exception but not wrapper in the second case? Or the second assert is to prove that the first WasTracked didn't change it back to false?